packages/block-library/src/cover/edit.js

```diff
@@ -316,6 +320,76 @@ function CoverEdit( {
 		allowedBlocks,
 		templateLock,
 	} = attributes;
+
+	const coverImage = useRef( {
+		url,
+		type: backgroundType,
+		isFeaturedImage: useFeaturedImage,
+	} );
+
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const media = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+	const mediaUrl = media?.source_url;
+
+	useEffect( () => {
+		// Save the currently set image
+		if ( useFeaturedImage && url ) {
+			coverImage.current = {
+				url,
+				type: backgroundType,
+				isFeaturedImage: false,
+			};
+		}
+
+		// If the useFeaturedImage attribute is set
+		// but there is no featured media on the post
+		// we reset the media
+		if ( useFeaturedImage && ! mediaUrl ) {
```

Comment:
It's unclear to me what we're doing here and why we need that?

Reply:
If a cover block is bound to the post's featured image but the post has no featured image or the featured image has been removed then we need to clear all the attributes that were set to point to the post's featured image.

Reply:
I guess my question is why these attributes were pointing to the post's featured image in the first place.

Reply:
Simplified media URL management so that we don't store featured image url in the attribute at all. Also removed this so that when the user selects some cover props they stick even if the post has no featured image set.